README.md

```diff
@@ -599,6 +599,11 @@ Starting from 23.10, request cancellation can be checked directly on the
 the TRITONSERVER_RESPONSE_COMPLETE_FINAL flag at the end of response is still
 needed even the request is cancelled.
 
+*Note:*: If `response_sender` is reused across multiple requests, the model must
```

Comment:
Can you elaborate on "`response_sender` is reused across multiple requests"? I thought each request has exactly one response sender?

Reply:
Apologize for the confusion. It is correct that each request has only one response sender. I think what I meant was specific to the case that the TRT-LLM team was experiencing: 
A separate thread is launched to handle the requests instead of spawning a new thread for each request, and the thread is terminated in the `finalize` stage. In this scenario, the last `response_sender` in the thread will not be cleaned up before model gets unloaded, hence the model can not be unloaded properly. (You can refer to the testing model [here](https://github.com/triton-inference-server/server/pull/7246/files#diff-23fac13db1595795a1151f62b4d6042e05fbd0757bff2bc211218b4b5a0517beR82) for more details)

On a second thought, I think we can remove this section since the complete final flag is required for both request rescheduling and cancellation. In this case, even with that specific thread implementation, the response_sender should clean up the response_factory when the complete final flag is set.

Let me know if it makes sense to you!

Reply:
Thanks for the clarification! It makes sense.

One last question:  Does the thread let the `response_sender` go out of scope, before the thread is terminated?

Reply:
For this particular example, I think the `response_sender` goes out of scope when the thread starts the termination by calling `thread.join()` or something like that. The issue was that `thread.join()` is called within the `finalize`, so the `response_sender` will only go out of scope after `finalize` is invoked. If manually doing `del response_sender` in the thread then the `response_factory` can be cleaned up.

Reply:
Shouldn't the thread let the `response_sender`(s) go out of scope as soon as they are no longer used, i.e. after sending response final flag, before `thread.join()` is called?

Reply:
Yes, I think that is the ideal way to manage response_sender. I believe one possible use case where response_sender does not go out of scope as soon as the final flag is sent is that, for example, the response_sender variable in this [model](https://github.com/triton-inference-server/server/pull/7246/files#diff-23fac13db1595795a1151f62b4d6042e05fbd0757bff2bc211218b4b5a0517beR94) will only be cleaned up when the next request is arrived and it gets reassigned to the new response_sender. This is when the destructor of the previous response_sender gets called. So unless users manually call `del response_sender` at the end of the thread, the last response_sender will not go out of scope until `thread.join()` is called.